modules/elasticsearch-impl/src/main/java/org/opencastproject/elasticsearch/impl/AbstractElasticsearchIndex.java

```diff
@@ -760,4 +792,24 @@ public List<String> getTermsForField(String field, String type) {
       return chuck(e);
     }
   }
+
+  /**
+   * Checks if the index service is reachable.
+   *
+   * @param elasticSearchUrl the index service URL
+   * @return true if the index service is reachable, false otherwise
+   */
+  private boolean isIndexServiceReachable(String elasticSearchUrl) {
+    try {
+      URL url = new URL(elasticSearchUrl);
+      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+      connection.setRequestMethod("HEAD");
+      connection.setConnectTimeout(3000);
+      connection.setReadTimeout(3000);
+      int responseCode = connection.getResponseCode();
+      return (200 <= responseCode && responseCode <= 399);
```

Comment:
- Why try doing an HTTP request and not use the library to try connecting to the search index?
- A 3xx response code doesn't really ensure that the connection would work, would it?

Reply:
Also, does this work with required authentication?